core/src/test/scala/unit/kafka/log/LogManagerTest.scala

```diff
@@ -115,18 +113,16 @@ class LogManagerTest {
     logManager.shutdown()
     logManager = spy(createLogManager(dirs))
     val brokenDirs = mutable.Set[File]()
-    doAnswer(new Answer[Try[File]] {
-      override def answer(invocation: InvocationOnMock): Try[File] = {
-        // The first half of directories tried will fail, the rest goes through.
-        val logDir = invocation.getArgument[File](0)
-        if (brokenDirs.contains(logDir) || brokenDirs.size < dirs.length / 2) {
-          brokenDirs.add(logDir)
-          Failure(new Throwable("broken dir"))
-        } else {
-          invocation.callRealMethod().asInstanceOf[Try[File]]
-        }
+    doAnswer { invocation =>
```

Comment:
nit: should we use `()` here? Do we have any non-written rules on lambdas for when `()` or `{}` should be used.

Reply:
Do you mean `doAnswer(invocation => {...`? If so, I would say `no`. The block syntax is more concise and generally recommended for cases like this.